activiti-services/activiti-services-events/src/main/java/org/activiti/services/events/MessageProducerCommandContextCloseListener.java

```diff
@@ -0,0 +1,48 @@
+package org.activiti.services.events;
+
+import java.util.List;
+
+import org.activiti.engine.impl.context.Context;
+import org.activiti.engine.impl.interceptor.CommandContext;
+import org.activiti.engine.impl.interceptor.CommandContextCloseListener;
+import org.activiti.services.api.events.ProcessEngineEvent;
+import org.springframework.messaging.support.MessageBuilder;
+
+public class MessageProducerCommandContextCloseListener implements CommandContextCloseListener {
+
+    public static final String PROCESS_ENGINE_EVENTS = "processEngineEvents";
+
+    private final ProcessEngineChannels producer;
+
+    public MessageProducerCommandContextCloseListener(ProcessEngineChannels producer) {
+        this.producer = producer;
+    }
+
+    @Override
+    public void closed(CommandContext commandContext) {
+        CommandContext currentCommandContext = Context.getCommandContext();
+        @SuppressWarnings("unchecked")
+        List<ProcessEngineEvent> events = (List<ProcessEngineEvent>) currentCommandContext
+                                                                                          .getAttribute(PROCESS_ENGINE_EVENTS);
+        if (events != null && events.size() != 0) {
+            producer.auditProducer().send(MessageBuilder.withPayload(events.toArray(new ProcessEngineEvent[events
```

Comment:
> Couldn't we use directly the list here instead of converting it to array?
I cannot use List because jackson couldn't deserialize List. 

> I think that following this change you will need to update the signatue of the method QueryConsumerChannelHandler.receive as well.

Thank you. I see. But, Can I try this fix on another branch and  pull request?